app/pages/sections/order-items/associated-services/delete/ui-tests/general.ui.test.js

```diff
@@ -0,0 +1,156 @@
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import content from '../manifest.json';
+import { nockAndErrorCheck, setState, authTokenInSession } from '../../../../../../test-utils/uiTestHelper';
+import { orderApiUrl } from '../../../../../../config';
+
+const pageUrl = 'http://localhost:1234/order/organisation/order-id/associated-services/delete/order-item-id/confirmation/write-on-time-associated-service/';
+
+const orderDescriptionMock = 'desc';
+
+const mocks = ({ postRoute = false }) => {
+  nock(orderApiUrl)
+    .get('/api/v1/orders/order-id/sections/description')
+    .reply(200, { description: orderDescriptionMock });
+
+  if (postRoute) {
+    nock(orderApiUrl)
+      .delete('/api/v1/orders/order-id/order-items/order-item-id')
+      .reply(204);
+  }
+};
+
+const defaultPageSetup = { withAuth: true, getRoute: true, postRoute: false };
+const pageSetup = async (setup = defaultPageSetup) => {
+  if (setup.withAuth) {
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+  }
+  if (setup.getRoute) {
+    mocks({ postRoute: setup.postRoute });
+  }
+};
+
+const getLocation = ClientFunction(() => document.location.href);
+
+fixture('delete-associated-service page - general')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    await nockAndErrorCheck(nock, t);
+  });
+
+test('when user is not authenticated - should navigate to the identity server login page', async (t) => {
+  nock('http://identity-server')
+    .get('/login')
+    .reply(200);
+
+  await pageSetup({ ...defaultPageSetup, withAuth: false, getRoute: false });
+  await t.navigateTo(pageUrl);
+
+  await t
+    .expect(getLocation()).eql('http://identity-server/login');
+});
+
+test('should render associated-service select page', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const page = Selector('[data-test-id="delete-catalogue-page"]');
+
+  await t
+    .expect(page.exists).ok();
+});
+
+test('should link to /order/organisation/order-id/associated-services for backLink', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const goBackLink = Selector('[data-test-id="go-back-link"] a');
+
+  await t
+    .expect(await extractInnerText(goBackLink)).eql(content.backLinkText)
+    .expect(goBackLink.getAttribute('href')).eql('/order/organisation/order-id/associated-services/order-item-id');
+});
+
+test('should render the title', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const title = Selector('h1[data-test-id="delete-catalogue-page-title"]');
+
+  await t
+    .expect(await extractInnerText(title)).eql('Delete write-on-time-associated-service from order-id?');
+});
+
+test('should render the description', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const description = Selector('h2[data-test-id="delete-catalogue-page-description"]');
+
```

Comment:
I can't see the template in this commit but the description should be a `p` not a `h2` with the classes `nhsuk-heading-l nhsuk-u-font-size-22` to make it look the same.

Reply:
This template is used in the form for all types - catalogue solutions, additional/associated services so - the change will require changing tests for all the calls and should be a new task/story.

Reply:
Ok, I will add a task for it on technical improvements, it is part of the accessibility audit.